github-upload-public-key/run.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+set -e
+
+CODER_ACCESS_URL="${CODER_ACCESS_URL}"
+CODER_OWNER_SESSION_TOKEN="${CODER_OWNER_SESSION_TOKEN}"
+GITHUB_EXTERNAL_AUTH_ID="${GITHUB_EXTERNAL_AUTH_ID}"
+
+if [ -z "$CODER_ACCESS_URL" ]; then
+  echo "No coder access url specified!"
+  exit 1
+fi
+
+if [ -z "$CODER_OWNER_SESSION_TOKEN" ]; then
+  echo "No coder owner session token specified!"
+  exit 1
+fi
+
+if [ -z "$GITHUB_EXTERNAL_AUTH_ID" ]; then
+  echo "No GitHub external auth id specified!"
+  exit 1
+fi
+
+echo "Fetching GitHub token..."
+GITHUB_TOKEN=$(coder external-auth access-token $GITHUB_EXTERNAL_AUTH_ID)
+if [ $? -ne 0 ]; then
+  echo "Failed to fetch GitHub token!"
+  exit 1
+fi
+if [ -z "$GITHUB_TOKEN" ]; then
+  echo "No GitHub token found!"
+  exit 1
+fi
+echo "GitHub token found!"
+
+echo "Fetching Coder public SSH key..."
+PUBLIC_KEY_RESPONSE=$(
+  curl -L -s \
+    -w "%%{http_code}" \
+    -H 'accept: application/json' \
+    -H "cookie: coder_session_token=$CODER_OWNER_SESSION_TOKEN" \
+    "$CODER_ACCESS_URL/api/v2/users/me/gitsshkey"
+)
+PUBLIC_KEY_RESPONSE_STATUS=$(tail -n1 <<< "$PUBLIC_KEY_RESPONSE")
+PUBLIC_KEY_BODY=$(sed \$d <<< "$PUBLIC_KEY_RESPONSE")
+
+if [ "$PUBLIC_KEY_RESPONSE_STATUS" -ne 200 ]; then
+  echo "Failed to fetch Coder public SSH key with status code $PUBLIC_KEY_RESPONSE_STATUS!"
+  echo "$PUBLIC_KEY_BODY"
+  exit 1
+fi
+
+PUBLIC_KEY=$(jq -r '.public_key' <<< "$PUBLIC_KEY_BODY")
+echo "Coder public SSH key found!"
+
+if [ -z "$PUBLIC_KEY" ]; then
+  echo "No Coder public SSH key found!"
+  exit 1
+fi
+
+echo "Fetching GitHub public SSH keys..."
+GITHUB_KEYS_RESPONSE=$(
+  curl -L -s \
+    -w "%%{http_code}" \
+    -H "Accept: application/vnd.github+json" \
+    -H "Authorization: Bearer $GITHUB_TOKEN" \
+    -H "X-GitHub-Api-Version: 2022-11-28" \
+    https://api.github.com/user/keys
+)
```

Comment:
I suggest we generalize to accept custom GitHub URLs for GitHub enterprises customers. 
We can do that by accepting an additional input and default to github.com as fallback. 

Bonus if we can auto detect the GitHub URL from the the configured external auth. 

cc: @Emyrk